stl/inc/xlocale

```diff
@@ -1989,33 +1988,123 @@ protected:
         _Adl_verify_range(_First2, _Last2);
         _Mid1 = _First1;
         _Mid2 = _First2;
-        for (;;) {
-            if (_Mid1 == _Last1) {
-                return ok;
+
+        // Handle 0 length string
+        if (_Mid1 == _Last1)
+            return ok;
+
+        // "C" -> Unicode direct translation
+        if (_Cvt._Isclocale) {

```

Comment:
does GCC/Clang on linux really widen by just casting like this? Additionally on linux wchar_t is utf32, so this makes a _lot_ more sense there. It's a little dubious on our platform. And again is such a massive behavior change I don't think we can do it.